### ISSUE TITLE
anchors with a property are clickable even when there is no value

### ISSUE THREAD

**Issue by MAKandCheese** [mavoweb/mavo#677]
Hello,

I found a couple small issues:

1) When a user leaves the property on an <a> tag blank in mavo, the link remains clickable. It just defaults to reload the page itself. It would be nice if mavo did not treat an absent value as a value when the user provided none. 
 
2) When a coder provides no property for the 'a' tag and there is no property inside the a tag, no text shows up at all. 

Codepen: https://codepen.io/MAKandCheese/pen/MWjqbvz?editors=1100

Thanks!

**Comment by LeaVerou** [comment]
Interesting. The Web itself allows empty links, and the empty href is resolved as relative to the current page, i.e. the same as `.`. Making links non-clickable in this case would make them externally inconsistent. On the other hand, I cannot think of any actual use cases where this behavior would be desirable (and if there are, one can always just specify `.`). 

Thoughts @DmitrySharabin @karger? 

The second issue is indeed confusing, and a separate problem. 

What behavior would have made sense to you in these two cases @MAKandCheese?

**Comment by DmitrySharabin** [comment]
I'm not sure whether it worth making all empty links unclickable and break the default browser behavior; links that follow nowhere (or simply reload a page), to my mind, are an evident sign that something is wrong with the code and/or data.

However, I believe we could make this behavior _possible_ if needed by letting authors specify, e.g., a class or an attribute on such anchors.

**Comment by karger** [comment]
Agreed that the right semantics for <a href> without value is a link 
that resolves to the current page.  But I think this first question can 
also be addressed by asking a different question: when a particular 
property is associated with an attribute, should an undefined value for 
the property correspond to the attribute having *no value* or to the 
attribute being *absent*?

**Comment by karger** [comment]
Agreed that the right semantics for <a href> without value is a link that resolves to the current page.  But I think this first question can also be addressed by asking a different question: when a particular property is associated with an attribute, should an undefined value for the property correspond to the attribute having *no value* or to the attribute being *absent*?

**Comment by LeaVerou** [comment]
Please note that if anchors with an empty `href` become non clickable, this means that these two have different behavior:

```html
<a href="">Foo</a>
<a href="" property="bar">Foo</a>
```

**Comment by karger** [comment]
So now we get to the question of whether an empty string denotes an absent value.  If I write `<a href="" property="bar">Foo</a>` that would seem to say that I am specifying that the property with this element is bar and its value is the empty string.  So since the property has a value, the empty string, that gets reflected back as the value of the href and has the standard html behavior---clickable and refers to current page.  On the other hand if I write `<a property="bar">Foo</a>` then I am saying this element is associated with the bar property but the bar property does not have a value.  Which could mean as I said about that there is no href attribute which means the element will not be clickable.  

And note this is consistent with the visible syntax:  `<a href="" property="bar">Foo</a>`  has an href attribute, so is clickable.  `<a property="bar">Foo</a>`  does not have one, so is not clickable.  The "surprise" happens because mavo chooses to *add* an href attribute where there isn't one, just because there is a property associated with the element.
